src/lib.rs

```diff
@@ -54,8 +54,9 @@ impl CallbackThreadPool {
         let mut thread_pool = Vec::with_capacity(thread_num);
         for _ in 0..thread_num {
             let receiver = cb_receiver.clone();
-            thread_pool.push(thread::spawn(move || loop {
-                if let Ok(msg) = receiver.recv() {
+            thread_pool.push(thread::spawn(move || {
+                // if cb_sender is released, break the while-loop and stop the thread
+                while let Ok(msg) = receiver.recv() {
```

Comment:
Practically speaking, this will only occur if `UPClientZenoh` has been dropped, right? Seems reasonable to me.

Reply:
You're right. While `UPClientZenoh` is dropped, the `cb_sender` will be released and also break the loop here to stop threads.